core/src/text/fontContext.cpp

```diff
@@ -40,35 +45,54 @@ void FontContext::unlock() {
     m_contextMutex.unlock();
 }
 
-bool FontContext::addFont(const std::string& _family, const std::string& _weight, const std::string& _style) {
+FontID FontContext::addFont(const std::string& _family, const std::string& _weight,
+                            const std::string& _style, bool _tryFallback) {
+
+    unsigned int dataSize = 0;
+    unsigned char* data = nullptr;
+    int font = FONS_INVALID;
 
     std::string fontKey = _family + "_" + _weight + "_" + _style;
-    if (m_fonts.find(fontKey) != m_fonts.end()) {
-        return true;
-    }
 
-    unsigned int dataSize;
-    unsigned char* data = nullptr;
+    auto it = m_fonts.find(fontKey);
+    if (it != m_fonts.end()) {
+        if (it->second < 0) {
+            goto fallback;
```

Comment:
The clang devs seems to care so much about goto that it warns about all impossible cases of uninitialized variables before goto statements that are not being used..

Wouldn't mind to repeat the 'fallback' case three times or write a function to wrap that part if we have any purists :)

You'll find another goto usage in the sceneloader refactoring where it is used to break out of a nested loop:
https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#Res-goto